File: cosmos-master/code/search/binary_search/binary_search.py
'''
Part of Cosmos by OpenGenus Foundation
'''

def binarySearch (arr, l, r, x):
 
    if r >= l:
 
        mid = l + (int)((r - l)/2)
 
        if arr[mid] == x:
            return mid
         
        elif arr[mid] > x:
            return binarySearch(arr, l, mid-1, x)
 
        else:
            return binarySearch(arr, mid+1, r, x)
 
    else:
        return -1
 
arr = [ 1, 2, 3, 5 ]
find = 3
 
print("Position of ",find," is ",binarySearch(arr, 0, len(arr)-1, find))

File: cosmos-master/code/search/binary_search/binary_search.c
#include <stdio.h>
/*
 * Part of Cosmos by OpenGenus Foundation
*/
int binarySearch(int arr[], int l, int r, int x)
{
   if (r >= l)
   {
        int mid = l + (r - l)/2;
 
        if (arr[mid] == x)  
            return mid;
         if (arr[mid] > x) 
             return binarySearch(arr, l, mid-1, x);
       
         return binarySearch(arr, mid+1, r, x);
   }
   return -1;
}
 
int main(void)
{
   int arr[] = {1, 2, 3, 5};
   int size = sizeof(arr)/ sizeof(arr[0]);
   int find = 3;
   printf("Position of %d is %d\n", find, binarySearch(arr, 0, size-1, find));
   return 0;
}